mlflow/utils/validation.py

```diff
@@ -345,6 +345,19 @@ def _validate_experiment_name(experiment_name):
         )
 
 
+def _validate_start_run_experiment_id(experiment_id):
```

Comment:
There is already a function `_validate_experiment_id` that performs a regex check on the `experiment_id` that is used in a different capacity (file store operations `_get_experiment(experiment_id)`, `_get_experiment_files(experiment_id)`, `_get_experiment_tag_path(experiment_id)`) that I didn't think would be worthwhile to combine as this is distinct validation logic for user-provided experiment_ids that can very rapidly throw based on invalid types.